qcodes/instrument/meta/meta_parameter.py

```diff
@@ -0,0 +1,174 @@
+from qcodes.instrument.group_parameter import Group
+from typing import Any, Dict, Optional, Tuple, Union
+from collections import namedtuple
+from qcodes.instrument.parameter import (
+    ParamDataType,
+    ParamRawDataType,
+    Parameter,
+    _BaseParameter
+)
+
+import logging
+
+_log = logging.getLogger(__name__)
+
+
+class MetaGroup(Group):
+    """Group of Meta parameters"""
+    def __init__(
+        self,
+        name: str,
+        parameters: Tuple['MetaParameter'],
+        parameter_names: Tuple[str] = None,
+        **kwargs
+    ):
+        super().__init__(parameters=list(parameters), single_instrument=False, **kwargs)
+        parameter_names = parameter_names or [_e.name for _e in parameters]
+        self._namedtuple = namedtuple(name, parameter_names)
+
+    def get_parameters(self) -> Any:
+        return self._namedtuple(*[_p.get() for _p in self.parameters.values()])
+
+    def _set_from_dict(self, calling_dict: Dict[str, ParamRawDataType]) -> None:
+        for name, p in list(self.parameters.items()):
+            p.set(calling_dict[name])
+
+
+class MetaParameter(_BaseParameter):
+    """
+    Meta parameter that returns one or more endpoint parameter values
+    
+    Args:
+        name: Meta parameter name
+        endpoints: One or more endpoint parameters to alias
+        endpoint_names: One or more endpoint names
+        unit: The unit of measure. Use ``''`` for unitless.
+        setter: Optional setter function to override
+            endpoint parameter setter. Defaults to None.
+    """
+    def __init__(
+        self,
+        name: str,
+        endpoints: Tuple[Parameter],
+        endpoint_names: Tuple[str],
+        unit: str = None,
+        setter: callable = None,
+        **kwargs
+    ):
```

Comment:
Why does the metaparameter not expose a label?  In this case label will simply be the same as name if there is nothing supplied. But this means that it is not possible to git it a nicer formatted label. 

name should be without spaces etc for use in generation of identifiers but label can be formatted with spaces etc

Reply:
Label is part of the Parameter interface but not the `_Baseparameter` interface but I think it makes sense to add it here

Reply:
Sounds good, thanks for the suggestion.